exporter/collector/googlemanagedprometheus/naming.go

```diff
@@ -28,7 +28,11 @@ func GetMetricName(baseName string, metric pmetric.Metric) (string, error) {
 	// Second, ad the GMP-specific suffix
 	switch metric.Type() {
 	case pmetric.MetricTypeSum:
-		return compliantName + getUnknownMetricSuffix(metric.Sum().DataPoints(), "/counter", "counter"), nil
+		if metric.Sum().IsMonotonic() {
+			return compliantName + getUnknownMetricSuffix(metric.Sum().DataPoints(), "/counter", "counter"), nil
+		}
+		// Non-monotonic sums are converted to GCM gauges
+		return compliantName + "/gauge", nil
```

Comment:
nit i'd consider monotonic the happy path and would swap which case ends up in the indented if block

Reply:
done